lib/webrick/httpresponse.rb

```diff
@@ -268,11 +287,10 @@ def setup_header() # :nodoc:
       elsif %r{^multipart/byteranges} =~ @header['content-type']
         @header.delete('content-length')
       elsif @header['content-length'].nil?
-        if @body.respond_to? :readpartial
```

Comment:
I think the logic here could be removed entirely, but it's basically a best effort mechanism to establish the `content-length` if possible, so that `connection: close` can be avoided. But both are semantically okay, it's just a little performance overhead using `connection: close` which honestly doesn't seem like a big issue for WEBrick as it's not expected to be super high performance.